### PR TITLE
Added connect models

### PR DESCRIPTION
**Pull request by Ripiter**

### CODE REVIEW

File: Source/Models/MQTTModels/ConnetPacket.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include "ConnectPayload.h"
+#include "ConnectVariableHeader.h"
+#include "ControllHeader.h"
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnetPacket.h]: Stavefejl

File: Source/Models/MQTTModels/ConnectPayload.h [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+#include <vector>
+
+namespace MQTTModels
+{
+	struct ConnectPayload
+	{		
+		int16_t ClientIDLenght;
+		std::vector<unsigned char> ClientID;
+
+		int16_t UsernameLenght;
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnectPayload.h]: Stavefejl

File: Source/Models/MQTTModels/ConnectPayload.h [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+#include <cstdint>
+#include <vector>
+
+namespace MQTTModels
+{
+	struct ConnectPayload
+	{		
+		int16_t ClientIDLenght;
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnectPayload.h]: Stavefejl

File: Source/Models/MQTTModels/ConnectVariableHeader.h [hunk 4]
```diff
@@ -0,0 +1,28 @@
+#pragma once
+#include <cstdint>
+#include <vector>
+
+namespace MQTTModels
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnectVariableHeader.h]: Follow namespace from diagram

File: Source/Models/MQTTModels/ConnectVariableHeader.h [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+	public:
+		int16_t NameLength;
+		std::vector<unsigned char> ProtocolName;
+		unsigned char Level;
+		ConnectFlag VariableLevel;
+		int16_t KeepAlive;
+	};
+
+	// move to its own header?
+	enum ConnectFlag {
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnectVariableHeader.h]: Own class

File: Source/Models/MQTTModels/ConnetPacket.h [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace MQTTModels
+{
+	struct ConnectPacket
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnetPacket.h]: ConnectPackage

File: Source/Models/MQTTModels/ConnetPacket.h [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+	struct ConnectPacket 
+	{
+
+	private:
+		ControllHeader m_controllHeader;
+		ConnectVariableHeader m_ConnectVariableHeader;
+		ConnectPayload m_ConnectPayload;
+	public:
+
+		inline ControllHeader GetConnectControllHeader() { return m_controllHeader; }
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnetPacket.h]: mangler const reference og controll = control

File: Source/Models/MQTTModels/ControllHeader.h [hunk 8]
```diff
@@ -0,0 +1,27 @@
+#pragma once
+
+namespace MQTTModels
+{
+	struct ControllHeader
+	{
+	public:
+		ControllFieldType PacketType;
```
Comment (Geefy) [inline Source/Models/MQTTModels/ControllHeader.h]: Controll = control

**Review by Geefy — CHANGES_REQUESTED** [review]

File: Source/Models/MQTTModels/ConnetPackage.h [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+	{
+		struct ConnectPackage
+		{
+		private:
+			ControlHeader& m_ControlHeader;
+			ConnectVariableHeader& m_ConnectVariableHeader;
+			ConnectPayload& m_ConnectPayload;
+		public:
+
+			inline const ControlHeader& GetConnectControllHeader() { return m_ControlHeader; }
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnetPackage.h]: Controll = Control

File: Source/Models/MQTTModels/ConnetPackage.h [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+			inline void SetConnectControllHeader(const ControlHeader& _controllHeader) { m_ControlHeader = _controllHeader; }
```
Comment (Geefy) [inline Source/Models/MQTTModels/ConnetPackage.h]: Controll = Control

File: Source/Models/MQTTModels/ConnectFlagType.h [hunk 11]
```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+namespace MQTT
+{
+	namespace Models
+	{
+		enum ConnectFlagType : int
```
Comment (blackout1471) [inline Source/Models/MQTTModels/ConnectFlagType.h]: The type should be unsigned char

**Review by Geefy — CHANGES_REQUESTED** [review]

File: Source/Models/MQTTModels/ConnectVariableHeader.h [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+#include <vector>
+#include "ConnectFlagType.h"
+
+namespace MQTT
+{
+	namespace Models
+	{
+		struct ConnectVariableHeader
+		{
+		public:
```
Comment (blackout1471) [inline Source/Models/MQTTModels/ConnectVariableHeader.h]: Public not needed

File: Source/Models/MQTTModels/ControlHeader.h [hunk 13]
```diff
@@ -0,0 +1,13 @@
+#pragma once
+#include "ControlPackageType.h"
+
+namespace MQTT
+{
+	namespace Models
+	{
+		struct ControlHeader {
+		public:
```
Comment (blackout1471) [inline Source/Models/MQTTModels/ControlHeader.h]: Public not needed

File: Source/Models/MQTTModels/ControlPackageType.h [hunk 14]
```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+namespace MQTT
+{
+	namespace Models
+	{
+		enum ControlPackageType : int
```
Comment (blackout1471) [inline Source/Models/MQTTModels/ControlPackageType.h]: Type should be unsigned char

**Comment by Geefy** [comment]
File paths are wrong (not right namespace) and Source/Models/MQTTModels/ConnetPackage.h has a spelling error, should be ConnectPackage.h

File: Source/Models/MQTTModels/ConnetPackage.h (hunk 9 above)
Comment (blackout1471) [reply Source/Models/MQTTModels/ConnetPackage.h]: Fixed it

File: Source/Models/MQTTModels/ConnetPackage.h (hunk 10 above)
Comment (blackout1471) [reply Source/Models/MQTTModels/ConnetPackage.h]: Fixed it

File: Source/Models/MQTTModels/ConnectFlagType.h (hunk 11 above)
Comment (blackout1471) [reply Source/Models/MQTTModels/ConnectFlagType.h]: Fixed

File: Source/Models/MQTTModels/ConnectVariableHeader.h (hunk 12 above)
Comment (blackout1471) [reply Source/Models/MQTTModels/ConnectVariableHeader.h]: fixed.

**Review by blackout1471 — APPROVED** [review]

**Review by Geefy — APPROVED** [review]

**Pull request merged**